### PR TITLE
lxc: Handle volume rename in `lxc storage volume move` command

### LINKED ISSUE #12344: Fails to rename custom storage volume
# Required information

* LXD version: `latest/edge`
* Storage backend in use: All backends seem to be affected

# Issue description

When renaming a custom storage volume with `lxc storage volume mv pool/vol1 pool/vol2` instead of `lxc storage volume rename pool vol1 vol2`, the client expects an operation to be returned. The command fails with `Error: not found`, however the actual volume gets renamed.

Renaming a volume returns a sync response which seems to be the root cause.

# Steps to reproduce

 1. `lxc storage volume create pool vol1`
 2. `lxc storage volume mv pool/vol1 pool/vol2 --debug` -> "Error: not found" when trying to GET the operation.

### PR DESCRIPTION
**Pull request by monstermunchkin**
When calling `lxc storage volume move` it internally always calls `lxc
storage volume copy`. This in itself is not a problem. However, the
`copy` handler expects LXD to always return an `OperationResponse`. This
doesn't happen though as LXD will return a `SyncResponse` if it detects
a move request.

This changes the `move` handler to call `lxc storage volume rename` if
it detects a rename request, instead of passing it to the `copy`
handler.

Fixes #12344

Signed-off-by: Thomas Hipp <thomas.hipp@canonical.com>

### CODE REVIEW

File: lxc/storage_volume.go [hunk 1]
```diff
@@ -1628,6 +1629,53 @@ func (c *cmdStorageVolumeMove) Run(cmd *cobra.Command, args []string) error {
...
+	dstVolName, dstVolPool := c.storageVolume.parseVolumeWithPool(dstPath)
+	if dstVolPool == "" {
+		return fmt.Errorf(i18n.G("No storage pool for target volume specified"))
+	}
+
+	// Rename volume if both remotes and pools of source and target are equal.
+	if srcRemote == dstRemote && srcVolPool == dstVolPool {
+		var args []string
+
+		if srcRemote != "" {
```
Comment (tomponline) [inline lxc/storage_volume.go:1668]: What does this bit do @monstermunchkin ?

Comment (monstermunchkin) [reply lxc/storage_volume.go:1668]: That's just to fill in the arguments for remote and pool correctly. `lxc storage volume rename` accepts `[<remote>:]<pool>` which is handled here.

Comment (tomponline) [reply lxc/storage_volume.go:1668]: Do we need an equivalent for the destination too then or does it apply to both?

Comment (monstermunchkin) [reply lxc/storage_volume.go:1668]: > Do we need an equivalent for the destination too then or does it apply to both?

Nope. `lxc storage volume rename` only takes one remote.

Comment (tomponline) [reply lxc/storage_volume.go:1668]: Great thanks. Please can you add a test for this new fix.

**Review by tomponline — APPROVED** [review]

**Pull request merged**